extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -536,10 +547,24 @@ export class AmpA4A extends AMP.BaseElement {
     //   - Chain cancelled => don't return; drop error
     //   - Uncaught error otherwise => don't return; percolate error up
     this.adPromise_ = viewerForDoc(this.getAmpDoc()).whenFirstVisible()
+        .then(() => {
+          checkStillCurrent(promiseId);
+          // See if experiment that delays request until slot is within
+          // renderOutsideViewport.
+          const resource = this.getResource();
+          // Within render outside viewport will not resolve if already within.
+          if (this.delayRequestEnabled && !resource.renderOutsideViewport()) {
+            return resource.whenWithinRenderOutsideViewport();
```

Comment:
Could you elaborate on the race condition?  I only wait on the whenWithinRenderOutsideViewport if renderOutsideViewport returns false (meaning within the first viewport will not wait).  But more importantly, our code assumes that onLayoutMeasure always executes before layoutCallback (otherwise adPromise_ would be null).   We check for this case at https://github.com/ampproject/amphtml/blob/master/extensions/amp-a4a/0.1/amp-a4a.js#L909 and I couldn't find an instance of it in error reporting.